sdk/lib/arbiter/arbiter.h

```diff
@@ -0,0 +1,255 @@
+#include <ndk/iofuncs.h>
+#include <ndk/kefuncs.h>
+#include <ndk/ldrfuncs.h>
+#include <ndk/mmfuncs.h>
+#include <ndk/obfuncs.h>
+#include <ndk/psfuncs.h>
+#include <ndk/rtlfuncs.h>
+#include <ndk/setypes.h>
+#include <ndk/sefuncs.h>
+#include <ndk/umfuncs.h>
+
+/* SEH support with PSEH */
+#include <pseh/pseh2.h>
+#endif
+
+typedef struct _ARBITER_ORDERING
+{
+    ULONGLONG Start;
+    ULONGLONG End;
+} ARBITER_ORDERING, *PARBITER_ORDERING;
+
+typedef struct _ARBITER_ORDERING_LIST
+{
+    USHORT Count;
+    USHORT Maximum;
+    PARBITER_ORDERING Orderings;
+} ARBITER_ORDERING_LIST, *PARBITER_ORDERING_LIST;
+
+typedef struct _ARBITER_ALTERNATIVE
+{
+    ULONGLONG Minimum;
+    ULONGLONG Maximum;
+    ULONG Length;
+    ULONG Alignment;
+    LONG Priority;
+    ULONG Flags;
+    PIO_RESOURCE_DESCRIPTOR Descriptor;
+    ULONG Reserved[3];
+} ARBITER_ALTERNATIVE, *PARBITER_ALTERNATIVE;
+
+typedef struct _ARBITER_ALLOCATION_STATE
+{
+    ULONGLONG Start;
+    ULONGLONG End;
+    ULONGLONG CurrentMinimum;
+    ULONGLONG CurrentMaximum;
+    PARBITER_LIST_ENTRY Entry;
+    PARBITER_ALTERNATIVE CurrentAlternative;
+    ULONG AlternativeCount;
+    PARBITER_ALTERNATIVE Alternatives;
+    USHORT Flags;
+    UCHAR RangeAttributes;
+    UCHAR RangeAvailableAttributes;
+    ULONG WorkSpace;
+} ARBITER_ALLOCATION_STATE, *PARBITER_ALLOCATION_STATE;
+
+typedef struct _ARBITER_INSTANCE *PARBITER_INSTANCE;
+
+typedef NTSTATUS
+(NTAPI * PARB_UNPACK_REQUIREMENT)(
+    _In_ PIO_RESOURCE_DESCRIPTOR IoDescriptor,
+    _Out_ PULONGLONG OutMinimumAddress,
+    _Out_ PULONGLONG OutMaximumAddress,
+    _Out_ PULONG OutLength,
+    _Out_ PULONG OutAlignment
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_PACK_RESOURCE)(
+    _In_ PIO_RESOURCE_DESCRIPTOR IoDescriptor,
+    _In_ ULONGLONG Start,
+    _Out_ PCM_PARTIAL_RESOURCE_DESCRIPTOR CmDescriptor
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_UNPACK_RESOURCE)(
+    _In_ PCM_PARTIAL_RESOURCE_DESCRIPTOR CmDescriptor,
+    _Out_ PULONGLONG Start,
+    _Out_ PULONG OutLength
+);
+
+typedef LONG
+(NTAPI * PARB_SCORE_REQUIREMENT)(
+    _In_ PIO_RESOURCE_DESCRIPTOR IoDescriptor
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_TEST_ALLOCATION)(
+    _In_ PARBITER_INSTANCE Arbiter,
+    _In_ PLIST_ENTRY ArbitrationList
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_RETEST_ALLOCATION)(
+    _In_ PARBITER_INSTANCE Arbiter,
+    _In_ PLIST_ENTRY ArbitrationList
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_COMMIT_ALLOCATION)(
+    _In_ PARBITER_INSTANCE Arbiter
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_ROLLBACK_ALLOCATION)(
+    _In_ PARBITER_INSTANCE Arbiter
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_BOOT_ALLOCATION)(
+    _In_ PARBITER_INSTANCE Arbiter,
+    _In_ PLIST_ENTRY ArbitrationList
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_QUERY_ARBITRATE)(
+    _In_ PARBITER_INSTANCE Arbiter
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_QUERY_CONFLICT)(
+    _In_ PARBITER_INSTANCE Arbiter
+);
+
+typedef NTSTATUS
+(NTAPI * PARB_ADD_RESERVED)(
+    _In_ PARBITER_INSTANCE Arbiter
```

Comment:
This function seems to also take a `PIO_RESOURCE_DESCRIPTOR` and a `PCM_PARTIAL_RESOURCE_DESCRIPTOR`